packages/metadata-tools/src/config-metadata.ts

```diff
@@ -0,0 +1,286 @@
+            endpointV2?: { address: string }
+            sendUln302?: { address: string }
+            lzExecutor?: { address: string }
+            sendUln301?: { address: string }
+            receiveUln301?: { address: string }
+            receiveUln302?: { address: string }
+        }[]
+        chainDetails?: {
+            chainType: string
+            chainKey: string
+            nativeChainId: number
+            chainLayer: string
+            chainStack?: string
+            nativeCurrency: {
+                name?: string
+                symbol: string
+                cgId?: string
+                cmcId: number
+                decimals: number
+            }
+            cgNetworkId?: string
+            shortName?: string
+            mainnetChainName?: string
+            name?: string
+        }
+        dvns?: {
+            [address: string]: {
+                version: number
+                canonicalName: string
+                id: string
+                deprecated?: boolean
+                lzReadCompatible?: boolean
+            }
+        }
+        rpcs?: { url: string; weight?: number }[]
+        addressToOApp?: {
+            [address: string]: {
+                id: string
+                canonicalName: string
+                type?: string
+            }
+        }
+        chainName: string
+        tokens?: {
+            [address: string]: {
+                symbol: string
+                cgId?: string
+                cmcId?: number
+                type: string
+                decimals: number
+                peggedTo?: {
+                    symbol: string
+                    chainName: string
+                    address: string
+                    programaticallyPegged?: boolean
+                }
+            }
+        }
+        chainKey: string
+    }
+}
+
+function getEndpointIdDeployment(eid: number, metadata: IMetadata) {
+    const srcEidString = eid.toString()
+    for (const objectKey in metadata) {
+        const entry = metadata[objectKey]
+
+        if (typeof entry?.deployments !== 'undefined') {
+            for (const deployment of entry.deployments) {
+                if (srcEidString === deployment.eid) {
+                    return deployment
+                }
+            }
+        }
+    }
+
+    throw new Error(`Can't find endpoint with eid: "${eid}",`)
+}
+
+function DVNsToAddresses(dvns: string[], chainKey: string, metadata: IMetadata) {
+    if (dvns.length === 0) {
+        return []
+    }
+
+    if (dvns[0]?.includes('0x')) {
+        return dvns.sort()
+    }
+
+    const dvnAddresses: string[] = []
+
+    if (!metadata[chainKey]?.dvns) {
+        throw new Error(`Can't find DVNs for chainKey: "${chainKey}".`)
+    }
+
+    const metadataDVNs = Object.entries(metadata[chainKey].dvns)
+
+    for (const dvn of dvns) {
+        for (const [dvnAddress, dvnDetails] of metadataDVNs) {
+            if (dvnDetails.canonicalName === dvn && !dvnDetails.lzReadCompatible) {
```

Comment:
We should add a conditional within this that checks if the DVN is `deprecated` and produce a warning on the CLI.

Reply:
Done.